### ISSUE TITLE
Chained method calls not indented

### ISSUE THREAD

**Issue by xixixao** [prettier/prettier#1401]
I am not sure whether this is a duplicate of existing chaining bugs:

Input (within 80):
```es6
const segments = AdamPropertyCountrySegmentsStore.getItems(propertyID)
  .map(resultMap => resultMap.toList());
```
output:
```es6
const segments = AdamPropertyCountrySegmentsStore.getItems(
  propertyID,
).map(resultMap => resultMap.toList());
```
Not sure why the first call is split across lines when it doesn't violate line limit (is it because it's counting the `.map(` call as well?).

Relatedly, this (within 80):
```es6
const segments = AdamPropertyCountrySegmentsStore.getItems(propertyID, a, b)
  .map(resultMap => resultMap.toList());
```
formats as
```es6
const segments = AdamPropertyCountrySegmentsStore.getItems(
  propertyID,
  a,
  b,
).map(resultMap => resultMap.toList());
```
which is not so bad since there are multiple args, but I'd probably still prefer the input.

**Comment by vjeux** [comment]
Good catch. Yeah, we should allow \n for map in this case.

**Comment by rattrayalex** [comment]
Well, for any chained method, right? Not just `map`?

**Comment by aretecode** [comment]
I saw this as well http://eslint.org/docs/rules/newline-per-chained-call

**Comment by xixixao** [comment]
@aretecode That is a good point, and I definitely see places in our codebase where inlining the chains makes code less readable. On the other hand, I do imagine DSLs where indenting every call wouldn't make sense. Compare

```es6
USD(100).GBP().add(20)
```
with
```es6
SomeNiceApi()
  .object('someID', id)
  .remove()
```
@vjeux This might be one of the few cases were we could leave onto the programmer whether they want to indent or inline (like we do for objects). I would prefer to find a good absolute rule, but I'm not sure it will be possible (we could consider the number of calls, their arguments, and the total length though).

**Comment by aretecode** [comment]
@xixixao for sure, that's why the rule allows things such as `'newline-per-chained-call': ['error', {ignoreChainWithDepth: 2}],`

**Comment by vjeux** [comment]
@xixixao I don't want to give people the choice here, but I do think that we can be more aggressive in expanding. There are precedent, for example we always expand function calls with `{}` and never inline them even if they fit.

**Comment by rattrayalex** [comment]
Agreed with @xixixao . Would probably be helpful to amass some examples of cases where it makes sense to keep them on the same line and derive rules (eg; fewer than three calls with fewer than two args each). 

Would probably also make sense to break all of them if the child breaks at all, no? Perhaps that how it already works?

**Comment by vjeux** [comment]
There's already a ton of tests that cover member chain in the codebase btw.

If you add 

```js
    !shouldMerge &&
```

in the condition for

```js
return group(oneLine);
```

this fixes the original problem but introduces a lot of unintended side effects. I would encourage you to play with the rules and try to come up with something that works better :)

**Comment by j-f1** [comment]
Current status:

**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEa4HMC2CMamAvJgIIAmAhvgAoBOEADnIzAJ4DCEArrEacAynkKw0wmBEZwAdLjgwAkvHxoAFC2ZsOnZQBEAlAB0omTHPw0WG2Wj4AbGAFkbZAHyZ7T1zbnSADIAlhgaRkYA3GZmkFAY2GJEJOTUdEys7Fy8AjBCogTJUjLyiipqmtqZeoYANJg09QBGpuaW1rY+zm4snt5wDt3+QaEw4VFmILUgrDDB6MigNIzMAO70ywhoyCA0AG4QwVRTuxjIAGY0jjjTTYw0YADWSsIsD8FQuMh5fHDTVBAwBcrjcQB8cBwmDQCDRgdc-iAAFZoAAeACF7k8XnQ4CEoHA4aD+DAWHwYAAmQkIt6MCE7Jo0JqcRzQE7aD4wADqRxgAAtkAAOAAM0yqOE59xYO20A3YewJ01kAEc+MFZFCYVTpjh8MFvoxftqPrhHHAAIp8CDwLUgGCM7lUPnIcnTPI0YKOY28fDWHZQaAKkB8HAAFUZ2yQl3hAF9o0A)

**Input:**
```jsx
const segments = AdamPropertyCountrySegmentsStore.getItems(propertyID)
  .map(resultMap => resultMap.toList());

const segments = AdamPropertyCountrySegmentsStore.getItems(propertyID, a, b)
  .map(resultMap => resultMap.toList());

```

**Output:**
```jsx
const segments = AdamPropertyCountrySegmentsStore.getItems(propertyID).map(
  resultMap => resultMap.toList()
);

const segments = AdamPropertyCountrySegmentsStore.getItems(
  propertyID,
  a,
  b
).map(resultMap => resultMap.toList());

```

**Comment by e-cloud** [comment]
@azz not just javascript but also typescript

**Comment by e-cloud** [comment]
```
this.userService.abc(kService.ka.it.id)
  .subscribe(
    user => {
      if (Array.isArray(user) && user.length > 0) {
        this.uService.get()
          .subscribe(
            userData => {
              if (userData.ut) {
                this.as.set('ut', userData.ut);
              }
              resolve(true);
            },
            err => {
              resolve(true);
            },
          );
      }
    },
    err => {
      reject(false);
    },
  );
```
the code above can be used for testing output.
